Introduces BaseDrawerActivity

poisdeux opened the pull request:
* BaseDrawerActivity enforces a common layout which allows us to more
     easily add new common UI elements
   * All activities incorporating a drawer should extend BaseDrawerActivity
     Currently CastActivity is the only exception.

Note that I also refactored PVRActivity although I am unable to test the PVR section. Could someone test this for me? Especially the drawer icon should be tested, to see if it changes correctly from hamburger to arrow and back.

SyncedSynapse commented on app/src/main/java/org/xbmc/kore/ui/sections/addon/AddonsActivity.java:
```diff
@@ -20,76 +20,48 @@
-        navigationDrawerFragment.setUp(R.id.navigation_drawer, (DrawerLayout) findViewById(R.id.drawer_layout));
-
-        Fragment fragment;
         if (savedInstanceState == null) {
```
oops
!=

poisdeux replied:
I don't understand. If the bundle is not null this activity is recreated because of a configuration change. In which the fragment does not need to be recreated, right?

SyncedSynapse replied:
Take a look at the next line of code that remains. Better check in your local copy instead of here, which is probably confusing you.

poisdeux replied:
Ah I see now. I already changed the code to fix the issues on device configuration changes and couldn't relate it to your comment.  Sorry about that.

poisdeux commented:
I noticed that setting the navigation icon state is not preserved on device configuration changes. :( Debugging/fixing this now.

poisdeux commented:
I think I fixed all issues with configuration changes. I renamed BaseDrawerActivity to BaseMediaActivity as we have some activities using a drawer that don't use the generic media layout.
Again, I was not able to test the PVR section. Hopefully that works as expected...
